File: sdk/elixir/lib/rocksky/stats.ex
defmodule Rocksky.Stats do
  @moduledoc "`app.rocksky.stats.*` endpoints."

  alias Rocksky.HTTP

  @doc "Per-user stats. Params: `:did`."
  def get_stats(client, params \\ []),
    do: HTTP.query(client, "app.rocksky.stats.getStats", params)

  @doc "Year-in-review (\"Wrapped\") data. Params: `:did`, `:year`."
  def get_wrapped(client, params \\ []),
    do: HTTP.query(client, "app.rocksky.stats.getWrapped", params)
end

File: sdk/clojure/src/rocksky/stats.clj
(ns rocksky.stats
  "Endpoints under app.rocksky.stats.*"
  (:require [rocksky.client :as c]))

(defn get-stats
  "Get aggregate stats for a user.

  Required: `:did`."
  [client {:keys [did]}]
  (c/query client :app.rocksky.stats.getStats {:did did}))

(defn get-wrapped
  "Get a user's year-in-review Wrapped stats.

  Required: `:did`. Optional: `:year`."
  [client {:keys [did year]}]
  (c/query client :app.rocksky.stats.getWrapped {:did did :year year}))
